pkg/controller/install/webhook.go

```diff
@@ -0,0 +1,43 @@
+package install
+
+import (
+	"fmt"
+
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+)
+
+func ValidWebhookRules(rules []admissionregistrationv1.RuleWithOperations) error {
+	for _, rule := range rules {
+		apiGroupMap := listToMap(rule.APIGroups)
+
+		// protect OLM resources
+		if contains(apiGroupMap, "*") {
+			return fmt.Errorf("Webhook rules cannot include all groups")
+		}
+
+		if contains(apiGroupMap, "operators.coreos.com") {
+			return fmt.Errorf("Webhook rules cannot include the OLM group")
+		}
+
+		// protect Admission Webhook resources
+		if contains(apiGroupMap, "admissionregistration.k8s.io") {
+			resourceGroupMap := listToMap(rule.Resources)
+			if contains(resourceGroupMap, "*") || contains(resourceGroupMap, "MutatingWebhookConfiguration") || contains(resourceGroupMap, "ValidatingWebhookConfiguration") {
```

Comment:
Might want to return an error specific to having `*` as a resource identifier: `Webhook rules cannot match all resources`.

Reply:
To be clear, Webhooks are allowed to match all resources in an APIGroup but they can't intercept Webhook resources.